### PR TITLE
Standardize Bazel setup.

### PR DESCRIPTION
**Pull request by smolkaj**
This makes it straightforward to depend on p4-constraints from within other Bazel projects.
Addresses issue #29.

### CODE REVIEW

**Comment by smolkaj** [comment]
@stefanheule: This is ready for review. Thanks :)

File: README.md [hunk 1]
```diff
@@ -150,30 +161,26 @@ You can also build p4-constraint in a Docker container, for example:
...
-which p4c  # This needs to succeed for golden tests to work.
-```
-You may want to use our [Docker container](#docker), which comes with p4c
-preinstalled.
-
 To run all golden tests, execute
 ```sh
 bazel test //e2e-test/...
 ```
+[Recall](building) that this will build p4c and requires
```
Comment (stefanheule) [inline README.md:180]: Does this link work? What does it link to? (not sure how to test this myself on github).

**Comment by stefanheule** [comment]
This is awesome, thank you so much!

File: README.md (hunk 1 above)
Comment (smolkaj) [reply README.md:180]: It links to the `## Building` section of the README. (To test, click ... at the top of the README.md diff and then "View File".)

Comment (smolkaj) [reply README.md]: And sure enough, the link was broken. It needed a '#'.
Good catch, thanks!

**Pull request merged**